pulp_docker/tests/functional/api/test_pull_content.py

```diff
@@ -0,0 +1,215 @@
+
+        * `Pulp #4460 <https://pulp.plan.io/issues/4460>`_
+        """
+        cls.cfg = config.get_config()
+        cls.client = api.Client(cls.cfg, api.page_handler)
+        cls.teardown_cleanups = []
+
+        with contextlib.ExitStack() as stack:
+            # ensure tearDownClass runs if an error occurs here
+            stack.callback(cls.tearDownClass)
+
+            # Step 1
+            _repo = cls.client.post(REPO_PATH, gen_repo())
+            cls.teardown_cleanups.append((cls.client.delete, _repo['_href']))
+
+            # Step 2
+            cls.remote = cls.client.post(
+                DOCKER_REMOTE_PATH, gen_docker_remote()
+            )
+            cls.teardown_cleanups.append(
+                (cls.client.delete, cls.remote['_href'])
+            )
+
+            # Step 3
+            sync(cls.cfg, cls.remote, _repo)
+            cls.repo = cls.client.get(_repo['_href'])
+
+            # Step 4
+            cls.publisher = cls.client.post(
+                DOCKER_PUBLISHER_PATH, gen_publisher()
+            )
+            cls.teardown_cleanups.append(
+                (cls.client.delete, cls.publisher['_href'])
+            )
+
+            # Step 5.
+            cls.publication = publish(cls.cfg, cls.publisher, cls.repo)
+            cls.teardown_cleanups.append(
+                (cls.client.delete, cls.publication['_href'])
+            )
+
+            # Step 6.
+            cls.distribution = cls.client.post(
+                DOCKER_DISTRIBUTION_PATH,
+                gen_distribution(publication=cls.publication['_href'])
+            )
+            cls.teardown_cleanups.append(
+                (cls.client.delete, cls.distribution['_href'])
+            )
+
+            # remove callback if everything goes well
+            stack.pop_all()
+
+    @classmethod
+    def tearDownClass(cls):
+        """Clean class-wide variable."""
+        for cleanup_function, args in reversed(cls.teardown_cleanups):
+            cleanup_function(args)
+
+    def test_api_returns_same_checksum(self):
+        """Verify that pulp serves image with the same checksum of remote.
+
+        1. Call pulp repository API and get the content_summary for repo.
+        2. Call dockerhub API and get blobsums for synced image.
+        3. Compare the checksums.
+        """
+        # Get local checksums for content synced from remote registy
+        checksums = [
+            content['digest'] for content
+            in get_content(self.repo)[DOCKER_CONTENT_NAME]
+        ]
+
+        # Assert that at least one layer is synced from remote:latest
+        # and the checksum matched with remote
+        self.assertTrue(
+            any(
+                [
+                    result['blobSum'] in checksums
+                    for result in get_docker_hub_remote_blobsums()
+                ]
+            ),
+            'Cannot find a matching layer on remote registry.'
+        )
+
+    def test_pull_image(self):
```

Comment:
@rochacbruno maybe this suggestion will be out of the scope if the testcase but please consider to add tests where the fetch of the content goes directly through API calls.
podman and docker are cli clients that already send some headers to the registry and have their own logic; a lot of people intergrate with the registries by directly sending API calls where they define accept headers.
The reason i am suggesting to add these tests is that we are sure we return correct content( if  availability of course) based on incoming headers. For example we would cover this bug https://pulp.plan.io/issues/4314

Reply:
this section might be helpful to understand how registry interprets incoming accept_headers and what it decided to return. https://docs.docker.com/registry/spec/manifest-v2-2/#backward-compatibility

Reply:
@rochacbruno, @ipanova, I filed another test case testing what was suggested here.

https://pulp.plan.io/issues/4531

Reply:
Thanks @ipanova your suggestions will come in new PRs